src/students/students.service.ts

```diff
@@ -26,4 +31,121 @@ export class StudentsService {
   getEmailsFromAddedStudents(addedStudents: Student[]): string[] {
     return addedStudents.map(student => student.user.email);
   }
+
+  // eslint-disable-next-line max-lines-per-function
+  public async getAllStudents(pageOptionsDto: PageOptionsDto): Promise<PageDto<StudentsResponse>> {
+    const queryBuilder = this.getAllStudentsQuery(pageOptionsDto);
+    const itemCount = await queryBuilder.getCount();
+    const { entities } = await queryBuilder.getRawAndEntities();
+    const students: StudentsResponse[] = [];
```

Comment:
po za tym, zamiast robić `forEach` to zrób sobie `map` na `students` i nie będziesz musiał robić `push`

```js
const students: StudentsResponse[] = entities.map(entity => ({
  studentId: entity.id,
  createdAt: entity.createdAt,
  details: {
    // ...
  },
  ```
  i tutaj świetnie uda się wrzucić metodę mapującą z odpowiedniego pliku